test/suites/projects.sh

```diff
@@ -1415,3 +1415,88 @@ test_projects_backups_volume() {
   lxc project delete foo
   lxc storage volume delete "${pool}" vol
 }
+
+test_projects_force_delete() {
+  pool="lxdtest-$(basename "${LXD_DIR}")"
+  create_object_storage_pool s3
+
+  echo "Create project with all features enabled."
+  lxc project create foo -c features.networks=true -c features.networks.zones=true -c features.images=true -c features.profiles=true -c features.storage.volumes=true -c features.storage.buckets=true
+
+  echo "Create storage volume in project."
+  lxc storage volume create "${pool}" custom/vol1 --project foo
+
+  echo "Create storage bucket in project."
+  lxc storage bucket create s3 bucket1 --project foo
+
+  echo "Create network ACL in project."
+  lxc network acl create acl1 --project foo
+
+  echo "Create network zone in project."
+  lxc network zone create zone1 --project foo
+
+  echo "Create profile in project."
+  lxc profile create profile1 --project foo
+
+  echo "Add image to project."
+  deps/import-busybox --project foo --alias testimage
+
+  uplink_network="uplink$$"
+  if ovn_enabled; then
+    echo "Create OVN uplink network."
+    setup_ovn
+
+    echo "Create a dummy physical network for use as an uplink."
+    ip link add dummy0 type dummy
+    lxc network create "${uplink_network}" --type=physical parent=dummy0
+
+    echo "Set OVN ranges."
+    lxc network set "${uplink_network}" ipv4.ovn.ranges=192.0.2.100-192.0.2.254
+    lxc network set "${uplink_network}" ipv6.ovn.ranges=2001:db8:1:2::100-2001:db8:1:2::254
+
+    echo "Set IP routes that include OVN ranges."
+    lxc network set "${uplink_network}" ipv4.routes=192.0.2.0/24
+    lxc network set "${uplink_network}" ipv6.routes=2001:db8:1:2::/64
+
+    echo "Create OVN network in project."
+    lxc network create foonet --type ovn --project foo network="${uplink_network}"
+  fi
+
+  echo "Check that regular delete fails on non-empty project."
+  ! lxc project delete foo || false
+
+  echo "Check force delete of non-existent project."
+  ! lxc project delete nonexistent --force --yes || false
+
+  echo "Check force delete of default project fails."
+  ! lxc project delete default --force --yes || false
+
+  echo "Create and start instance."
+  lxc launch testimage c1 --project foo -s "${pool}"
+
+  echo "Force delete project."
+  lxc project delete foo --force --yes
+
+  echo "Check project is deleted."
+  ! lxc project show foo || false
+
+  echo "Clean up OVN parent network."
+  if ovn_enabled; then
+    lxc network delete "${uplink_network}"
```

Comment:
I presume that this acts as the validation that `foonet` was deleted along with the project. I think that having that network being used by `c1` would also make sure that even networks being used are deleted.